### PR TITLE
LPS-121149 merge evaluation results with last edited values

### PR DESCRIPTION
**Pull request by linolaoj**
Related Issue: https://issues.liferay.com/browse/LPS-121149

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hey @linolaoj, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by linolaoj** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/evaluation.es.js [hunk 1]
```diff
@@ -136,6 +142,7 @@ export const mergePages = (
 					sourceField.displayErrors || field.fieldName === fieldName,
 				editingLanguageId,
 				valid: field.valid !== false,
+				value: value,
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/evaluation.es.js]: @rijema and @linolaoj, please use a ternary operator here to save the declaration of variable `value`.

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js [hunk 2]
```diff
@@ -117,13 +121,20 @@ export default function fieldChange({
 						if (fieldInstance.isDisposed()) {
 							return;
 						}
+						const mergedPages = mergePages(
+							defaultLanguageId,
+							editingLanguageId,
+							fieldName,
+							evaluatedPages,
+							lastEditedPages
+						);
```
[the remark this answers is not in the document]
Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js:130]: Hi @natocesarrego ,  on evaluation.es.js the pages are passed as parameter before the server response, so when the merge happens there the field value might already be different (the server might take a couple of seconds on a slow connection). This merge was included to use the most updated values from the page.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js [hunk 3]
```diff
@@ -47,6 +47,8 @@ const UPDATE_DELAY_MS = 50;
 
 const debounceFn = debounce((fn) => fn(), UPDATE_DELAY_MS);
 
+let lastEditedPages = [];
```
[the remark this answers is not in the document]
Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js:50]: @natocesarrego , in our tests we could not get the expected behavior using the editedPages.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/evaluation.es.js (hunk 1 above)
Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/evaluation.es.js]: @rijema is going to resend

**Comment by linolaoj** [comment]
resent at https://github.com/liferay-forms/liferay-portal/pull/11

**Pull request closed without merging**